### PR TITLE
[IMP] Export Wizard New Look

### PR DESCRIPTION
**Pull request by rba-odoo**
hello,
I have improved design using bootstrap in export wizard, changed Export Wizard Dialogue box class to Widget class.

More Details:
Task: https://www.odoo.com/web#id=8651&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.mpua9w4G7VMVj4tD

thanks
rba-odoo

### CODE REVIEW

File: addons/web/static/src/js/data_export.js [hunk 1]
```diff
@@ -19,66 +19,96 @@ instance.web.DataExport = instance.web.Dialog.extend({
...
+            }else {
+                self.$el.find('#field-tree-structure').remove();
+                var import_comp = "";
+                self.rpc("/web/export/get_fields", {
+                    model: self.action.params.model,
+                    import_compat: !!import_comp,
+                }).done(function (records) {
+                    got_fields.resolve();
+                    self.on_show_data(records);
+                });
```
Comment (vme-odoo) [inline addons/web/static/src/js/data_export.js]: rethink of code inside if/else.

File: addons/web/static/src/js/data_export.js [hunk 2]
```diff
@@ -147,28 +176,19 @@ instance.web.DataExport = instance.web.Dialog.extend({
         _(field_list).each(function (field) {
             export_node.append(new Option(field.label, field.name));
         });
+        this.$el.find(".oe_export_file").removeAttr("disabled");
+        this.$el.find(".oe_button_unable").removeAttr("disabled");
```
Comment (vme-odoo) [inline addons/web/static/src/js/data_export.js]: this enable/disable several times use, you can define a method.

File: addons/web/static/src/js/data_export.js [hunk 3]
```diff
@@ -19,68 +19,92 @@ instance.web.DataExport = instance.web.Dialog.extend({
...
+            this, 'ir.exports', this.action.params.context);
     },
     start: function() {
         var self = this;
         this._super.apply(this, arguments);
+        var export_fields = this.$("#fields_list option").map(function() {
+            return $(this).val();
+        }).get();
 
+        domain=this.action.params.domain;
```
Comment (vme-odoo) [inline addons/web/static/src/js/data_export.js]: Why "domain" as global? and no space between assignation!!

File: addons/web/static/src/js/data_export.js [hunk 4]
```diff
@@ -19,68 +19,92 @@ instance.web.DataExport = instance.web.Dialog.extend({
...
+        self.$el.find(".oe_model_name").text(this.model_name);
```
Comment (vme-odoo) [inline addons/web/static/src/js/data_export.js]: set from template.

File: addons/web/static/src/js/views.js [hunk 5]
```diff
@@ -1405,7 +1405,25 @@ instance.web.View = instance.web.Widget.extend({
     do_search: function(domain, context, group_by) {
     },
     on_sidebar_export: function() {
-        new instance.web.DataExport(this, this.dataset).open();
+        var self= this;
+        var get_selected_ids = 0
+        if (!(get_selected_ids=self.get_selected_ids().length)) {
```
Comment (vme-odoo) [inline addons/web/static/src/js/views.js]: why assign value to *get_selected_ids* in IF condition?

File: addons/web/static/src/js/views.js [hunk 6]
```diff
@@ -1405,7 +1405,25 @@ instance.web.View = instance.web.Widget.extend({
...
+                type: 'ir.actions.client',
+                tag: 'export',
+                flags: {new_window: true},
+                params: {
+                    model: self.dataset.model,
+                    dataset: self.dataset,
+                    domain: self.get_active_domain(),
+                    selected_ids: get_selected_ids,
+                    context: self.getParent().action.context,
+                    view: self.ViewManager.action.name,
```
Comment (vme-odoo) [inline addons/web/static/src/js/views.js]: Most of times we get value, but what if don't?

File: addons/web/static/src/js/views.js [hunk 7]
```diff
@@ -1409,7 +1409,23 @@ instance.web.View = instance.web.Widget.extend({
...
+        }
+        this.do_action({
+            type: 'ir.actions.client',
+            tag: 'export',
+            flags: {new_window: true},
+            params: {
+                model: this.dataset.model || "",
+                dataset: this.dataset || [],
+                domain: this.get_active_domain() || [],
+                selected_ids: selected_ids || "",
```
Comment (vme-odoo) [inline addons/web/static/src/js/views.js]: seletected_ids || []

**Comment by rba-odoo** [comment]
@vme-odoo Done.! All required changes.!!

**Pull request closed without merging**